### ISSUE TITLE
Port to GTK4+libadwaita

### ISSUE THREAD

**Issue by TheEvilSkeleton** [jeffshee/hidamari#64]
Any plans on porting to GTK4+libadwaita?

**Comment by jeffshee** [comment]
@TheEvilSkeleton Actually, I'm not sure. For the GUI part (the control panel that write the config file), yes it's possible. But for the window that display the video wallpaper, it has some technical difficulties.

- GTK4 doesn't have a way to embed vlc by design. We could switch to GStreamer (by using Gtk.MediaFile), but that requires some reimplantation and I'm not too sure with the performance and stability.

**Comment by VarLad** [comment]
@jeffshee Currently it uses mpv right? I can't get the hw acceleration to work with Hidamari (installed from Flathub) + MPV :/
Any way to force the `--hwdec=vaapi` params?
I found the same issue in Celluloid Flatpak but loading an mpv config file with `hwdec=vaapi` solved that for me

**Comment by jeffshee** [comment]
@VarLad No, it uses VLC. You should make a new issue since your topic is different from OP. I will move yours over there.

**Comment by VarLad** [comment]
@jeffshee I'd guess Gstreamer would be a lot lighter of an alternative to VLC?
and since this is Linux-only, I'd think Gstreamer would be a better choice.

**Comment by jeffshee** [comment]
@VarLad It might be. However, Gstreamer integration in Gtk4 seems half-baked, based on my experience. If we really want to switch to Gstreamer, I think we should first make a minimal Gstreamer video player (in Gtk4 ofcuz) and do some unit testing.
